File: playground/javascript/node/recommend.ts
import { ApiError } from '@algolia/client-common';
import { recommendClient } from '@algolia/recommend';

const appId = process.env.ALGOLIA_APPLICATION_ID || '**** APP_ID *****';
const apiKey = process.env.ALGOLIA_SEARCH_KEY || '**** SEARCH_API_KEY *****';

const searchIndex = process.env.SEARCH_INDEX || 'test_index';
const searchQuery = process.env.SEARCH_QUERY || 'test_query';

// Init client with appId and apiKey
const client = recommendClient(appId, apiKey);

async function testGetRecommendations() {
  try {
    const res = await client.getRecommendations({
      requests: [
        {
          indexName: searchIndex,
          model: 'bought-together',
          objectID: searchQuery,
          threshold: 0,
        },
      ],
    });

    console.log('[OK] GetRecommendations', res);
  } catch (e) {
    if (e instanceof ApiError) {
      return console.log(`[${e.status}] ${e.message}`, e.stackTrace);
    }

    console.log('[ERROR] GetRecommendations', e);
  }
}

async function testRecommend() {
  await testGetRecommendations();
}

testRecommend();

File: playground/go/recommend.go
package main

import (
	"context"
	"fmt"

	"github.com/algolia/algoliasearch-client-go/v4/algolia/next/recommend"
)

func testRecommend(ctx context.Context, appID, apiKey string) int {
	recommendClient, err := recommend.NewClient(appID, apiKey)
	if err != nil {
		panic(err)
	}

	/*
		params := []recommend.RecommendationsRequest{
			recommend.RecommendationRequestAsRecommendationsRequest(recommend.NewRecommendationRequest(recommend.RECOMMENDATIONMODELS_BOUGHT_TOGETHER, "test_query", "test_index", 0)),
		}
	*/
	// alternative way to create the payloads, a similar approach can be used with any of the other clients
	params := []recommend.RecommendationsRequest{{
		BoughtTogetherQuery: &recommend.BoughtTogetherQuery{
			Model:     recommend.FBT_MODEL_BOUGHT_TOGETHER,
			ObjectID:  "test_query",
			IndexName: "test_index",
			Threshold: 0,
		},
	}}

	searchResponse, err := recommendClient.GetRecommendations(ctx, params)
	if err != nil {
		fmt.Printf("request error with SearchSingleIndex: %v\n", err)
		return 1
	}

	printResponse(searchResponse)

	return 0
}
